src/crimson/os/cyanstore/cyan_store.cc

```diff
@@ -416,6 +416,7 @@ seastar::future<> CyanStore::do_transaction(CollectionRef ch,
       t.get_on_applied_sync()}) {
     if (i) {
       i->complete(0);
+      delete i;
```

Comment:
why?

Reply:
alien store need keep context alive (for alienstore OnCommit promise get_future),  so in Context.h  comment out delete this,  if I identify cyan_store from alien store, I need add more namespace, so here treat them the same. 
#ifndef WITH_SEASTAR
    delete this;
#endif

Reply:
Wait, so with alien, Context::complete doesn't delete the context?  I think that's incredibly dangerous.  If anyone modifies BlueStore to use a context for a purpose other than the exposed interface, it'll introduce a memory leak.  Is there really no better way to do this?  Why can't you wrap the context before passing it to bluestore?